### PR TITLE
add gradle test with github actions

### PR DESCRIPTION
**Pull request by sethharling**
-automatically performs a build and java compile when code is either a) pushed to the main branch or b) a pull request is created for the main branch

-also we can add a cool badge to our readme file!

### CODE REVIEW

**Comment by sethharling** [comment]
forgot to make the ./gradlew command executable.

**Review by snowjtSD — APPROVED** [review]
Nice job man! Looks good. Looking forward to testing when it's merged.

**Comment by sethharling** [comment]
looks like the .gitignore had a conflict with the vscode settings and made it ignore the gradle-wrapper.properties file. will merge and then add later to avoid any unwanted uploads from vscode.

File: .gitignore [hunk 1]
```diff
@@ -158,3 +158,13 @@ dist
 npm-debug.log*
 yarn-debug.log*
 yarn-error.log*
+
+# Ignore Gradle project-specific cache directory
+.gradle
```
Comment (chillyssa) [inline .gitignore:163]: based on this declaration shouldn't build.gradle and settings.gradle be excluded from this merge?

**Review by chillyssa — COMMENTED** [review]
So I think we need to look at the way our .gitignore is currently set up and make some modifications /clarifications now that our tech stack is growing. I am not sure some of these files should be part of this merge.

**Review by chillyssa — APPROVED** [review]
I am seeing that the tests are passing, great job @sethharling getting this working. 

Approving for now but early next sprint I think we definitely want to make sure our .gitignore is setup for all technologies we are currently using and making sure they all play nice together/don't override other files somehow!

**Pull request merged**